src/common/components/header/index.js

```diff
@@ -7,7 +7,7 @@ import styles from './header.css';
 
 const wordmark = 'https://assets.ubuntu.com/v1/d45097a4-snapcraft.io-logotype.svg';
 
-class Header extends Component {
+export class Header extends Component {
```

Comment:
I've read this thread, the IRC conversation and [the article](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0#.5vryhppki) @katiefenn mentioned.

Since I'm still learning React, I don't have a specific opinion about this exact issue, but I do definitely believe that both encapsulation and a sensible separation of concerns are terribly important to try to prevent your codebase from growing into a spaghetti monster.

I very much like the sound of separating "Container" and "Presentational" logic, and also of strictly exporting only one thing from each component, and of the idea that we should be doing our thinking up-front to save future developers from having to think.

Trying to `export` more than one class sounds like it may be effectively trying to use global scope to make your life easier today. @katiefenn's suggestion on passing in properties instead, sounds similar to an example of [dependency injection](https://en.wikipedia.org/wiki/Dependency_injection), which is a widely accepted pattern for keeping neat, encapsulated units which are easier to test.

It is true that this can lead you to sometimes end up injecting an inordinately large number of dependencies. But this can be mitigated by sensibly breaking up your functions into smaller components, and then using [composition over inheritance](https://en.wikipedia.org/wiki/Composition_over_inheritance).

As I said, I don't understand enough to have a strong opinion about this specific piece of code, but since we are no longer under the gun, I do think it's worth taking the time to come up with patterns which ensure that concerns are separated, encapsulation is clean, and logical structures are kept as simple as possible.

If you do happen to come up with good rules of thumb, feel free to encode them into law by proposing a "javascript.md" or "react.md" into the [webteam practices coding folder](https://github.com/ubuntudesign/webteam-practices/tree/master/coding).